src/Integration.Vsix/ErrorList/ISonarErrorListDataSource.cs

```diff
@@ -25,7 +25,7 @@ namespace SonarLint.VisualStudio.Integration.Vsix
     /// </summary>
     internal interface ISonarErrorListDataSource
     {
-        void RefreshErrorList();
+        void RefreshErrorList(SnapshotFactory factory);
```

Comment:
Tell the Error List that it needs to refresh a single factory, rather than have it check all of the factories.

Actually, the main reason for this change is so that when it comes to notifying the taggers that something has changed (in a future PR), we won't need to notify the taggers of all open documents. We'll be able to notify only the taggers that deal with files that have issues in the factory.